everything/reports/orders.md

```diff
@@ -0,0 +1,543 @@
+
+The Order Approval Duration graph provides information on how long it takes the OMS to approve orders. It has two main attributes: order volume and average approval duration in minutes. This Graph helps in tracking the average approval duration over time. This helps in identifying any deviations from expected performance levels and taking corrective actions promptly.
+
+#### Field Header
+
+| Field Header                      | Description                                                                                         | HC Entity                 |
+|-----------------------------------|-----------------------------------------------------------------------------------------------------|---------------------------|
+| Order Volume                      | This refers to the number of orders received by the system over a specific period.                   | OrderHeader.ORDER_ID     |
+| Average Approval Duration (Minutes) | This is the average amount of time it takes for the OMS to approve an order.                         | Calculated difference of OrderHeader.ORDER_DATE and OrderStatus.STATUS_DATETIME(For “APPROVED” status) divided by 60 to display in minutes |
+
+#### SQL Query to Generate Order Approval Duration Graph
+
+```sql
+SELECT DATE(`ENTRY_DATE`) AS `ENTRY_DATE`,
+       AVG(`duration in sec`) / 60 AS `Average approval duration in minutes`,
+       count(DISTINCT `Order Id`) AS `Orders Volume`
+FROM
+  (select oh.ORDER_ID "Order Id",
+          oh.ORDER_NAME "Shopify Order Id",
+          TIME_FORMAT(SEC_TO_TIME(TIMESTAMPDIFF(SECOND, oh.ORDER_DATE, max(os.status_datetime))), "%H:%i:%s") "Duration",
+          TIMESTAMPDIFF(SECOND, oh.ORDER_DATE, max(os.status_datetime)) "duration in sec",
+          oh.ENTRY_DATE,
+          oh.ORDER_DATE "Created At",
+          max(os.status_datetime) "Approved At"
+   from order_header oh
+   join order_status os on os.ORDER_ID = oh.ORDER_ID
+   and os.status_id = "ORDER_APPROVED"
+   where os.status_datetime
+   group by oh.ORDER_ID
+   order by oh.ORDER_ID desc) AS virtual_table
+WHERE `Created At` >= STR_TO_DATE('2024-04-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+  AND `Created At` < STR_TO_DATE('2024-05-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+  AND `ENTRY_DATE` >= STR_TO_DATE('2024-04-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+  AND `ENTRY_DATE` < STR_TO_DATE('2024-05-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+GROUP BY DATE(`ENTRY_DATE`)
+ORDER BY `Average approval duration in minutes` DESC
+LIMIT 10000;
+```
+
+#### Data Selection
+
+The SQL query begins by selecting specific data from two tables: `order_header` and `order_status`. These tables likely contain information about orders and their statuses.
+
+#### Defining Criteria
+
+The query filters the data to include only orders with a specific status, in this case, orders that are approved. This is achieved by specifying the condition `os.status_id = 'ORDER_APPROVED'`.
+
+#### Joining Relevant Tables
```

Comment:
why are we using headings here, while we have used the bold list in others

Reply:
Done